src/Rotations.jl

```diff
@@ -0,0 +1,148 @@
+@doc doc"""
+    Rotations{N} <: Manifold
+
+Special orthogonal manifold $\mathrm{SO}(n)$ represented by $n\times n$
+real-valued orthogonal matrices with determinant $+1$.
+
+# Constructor
+
+    Rotations(n, retract! = rectract_qr!)
+
+generates the $\mathrm{SO}(n) \subset \mathbb R^{n\times n}$
+"""
+struct Rotations{N, TRetr<:Function} <: Manifold
+    retract!::TRetr
+end
+Rotations(n::Int, retract!::TRetr = retract_qr!) where TRetr = Rotations{n,TRetr}(retract!)
+
+@doc doc"""
+    dimension(S::Rotations)
+
+returns the dimension of the manifold $\mathrm{SO}(n)$, i.e. $\frac{n(n-1), 2}$.
+"""
+dimension(S::Rotations{N}) where {N} = div(N*(N-1), 2)
+
+@doc doc"""
+    inner(S,x,w,v)
+
+compute the inner product of the two tangent vectors `w,v` from the tangent
+plane at `x` on the special orthogonal space `S=`$\mathrm{SO}(n)$ using the
+restriction of the metric from the embedding, i.e. $ (v,w)_x = \frac{1,2}v^\mathrm{T}w $.
+"""
+inner(S::Rotations, x, w, v) = dot(w, v)/2
+
+project_tangent!(S::Rotations, w, x, v) = w .= (transpose(v).-v)./2
+
+function exp!(S::Rotations, y, x, v)
```

Comment:
Here we are assuming the tangent `v` is a matrix, but above we assumed it was a vector?